clientEncryptionToCPKNMigrationSamples/ClientSideKeyVaultKeyToCustomerManagedKey/src/main/java/Migration.java

```diff
@@ -0,0 +1,142 @@
+                                                     String storageAccount, String sharedKeyCred, String containerName,
+                                                     String blobName, String blobDecryptName, String keyVaultUrl, String keyname,
+                                                     String keyWrapAlgorithm, String path) {
+        String storageAccountUrl = "https://" + storageAccount + ".blob.core.windows.net";
+
+        // Setting encryptedKeyClient with key vault key
+        BlobClient blobClient = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .blobName(blobName)
+                .buildClient();
+        TokenCredential cred = new ClientSecretCredentialBuilder()
+                .clientSecret(clientSecret)
+                .tenantId(tenantId)
+                .clientId(clientId)
+                .build();
+        KeyClient keyClient = new KeyClientBuilder()
+                .vaultUrl(keyVaultUrl)
+                .credential(cred)
+                .buildClient();
+        KeyVaultKey rsaKey = keyClient.getKey(keyname);
+        EncryptedBlobClient encryptedBlobClient = new EncryptedBlobClientBuilder()
+                .key(createAsyncKey(rsaKey, cred), keyWrapAlgorithm)
+                .blobClient(blobClient)
+                .buildEncryptedBlobClient();
+
+        // Downloading encrypted blob, blob is decrypted upon download
+        encryptedBlobClient.downloadToFile(path + blobDecryptName);
+    }
+
+    /**
+     * Reuploads blob with server-side encryption using a customer-managed key
+     */
+    private static void encryptCustomerManagedKey(String storageAccount, String sharedKeyCred, String containerName,
+                                                  String blobDecryptName,
+                                                  String encryptionScope, String path) {
+        String storageAccountUrl = "https://" + storageAccount + ".blob.core.windows.net";
+        // Creating blob client for reuploading
+        BlobClientBuilder blobClientBuilder = new BlobClientBuilder()
+                .endpoint(storageAccountUrl)
+                .credential(new StorageSharedKeyCredential(storageAccount, sharedKeyCred))
+                .containerName(containerName)
+                .encryptionScope(encryptionScope)
+                .blobName(blobDecryptName);
+        BlobClient blobClientDecrypted = blobClientBuilder.buildClient();
+
+        // Uploading file to server
+        blobClientDecrypted.uploadFromFile(path + blobDecryptName, true);
+    }
+
+    /**
+     * Cleans up temp files created during decryption
+     */
+    private static void cleanup(String blobDecryptName, String path) {
+        // Cleaning up by deleting local save of encrypted blob
+        File localFile = new File(path + blobDecryptName);
```

Comment:
https://stackoverflow.com/questions/34459486/joining-paths-in-java . Better to use `Paths` to join paths (might be OS dependent).
Please make this change in other places you join paths (i.e. where you use `\\` literals in string).